doc/source/data/key-concepts.rst

```diff
@@ -4,108 +4,103 @@
+You can manually specify the number of read tasks, but the final parallelism is always capped by the number of files in the underlying datastream.
 
-For an in-depth guide on creating datasets, read :ref:`Creating Datasets <creating_datasets>`.
+For an in-depth guide on creating datastreams, read :ref:`Creating Datastreams <creating_datastreams>`.
 
 Transforming Data
 =================
 
-Datasets uses either Ray tasks or Ray actors to transform data blocks. By default, Datasets uses tasks.
+Datastream uses either Ray tasks or Ray actors to transform data blocks. By default, it uses tasks.
 
 To use Actors, pass an :class:`ActorPoolStrategy` to ``compute`` in methods like
-:meth:`~ray.data.Dataset.map_batches`. :class:`ActorPoolStrategy` creates an autoscaling
+:meth:`~ray.data.Datastream.map_batches`. :class:`ActorPoolStrategy` creates an autoscaling
 pool of Ray actors. This allows you to cache expensive state initialization
 (e.g., model loading for GPU-based tasks).
 
-.. image:: images/dataset-map.svg
+.. image:: images/datastream-map.svg
    :align: center
 ..
   https://docs.google.com/drawings/d/12STHGV0meGWfdWyBlJMUgw7a-JcFPu9BwSOn5BjRw9k/edit
 
-For an in-depth guide on transforming datasets, read :ref:`Transforming Datasets <transforming_datasets>`.
+For an in-depth guide on transforming datastreams, read :ref:`Transforming Datastreams <transforming_datastreams>`.
 
 Shuffling Data
 ==============
 
-Operations like :meth:`~ray.data.Dataset.sort` and :meth:`~ray.data.Dataset.groupby`
-require blocks to be partitioned by value or *shuffled*. Datasets uses tasks to shuffle blocks in a map-reduce
+Operations like :meth:`~ray.data.Datastream.sort` and :meth:`~ray.data.Datastream.groupby`
+require blocks to be partitioned by value or *shuffled*. Datastream uses tasks to shuffle blocks in a map-reduce
 style: map tasks partition blocks by value and then reduce tasks merge co-partitioned
 blocks.
 
-Call :meth:`~ray.data.Dataset.repartition` to change the number of blocks in a :class:`~ray.data.Dataset`.
+Call :meth:`~ray.data.Datastream.repartition` to change the number of blocks in a :class:`~ray.data.Datastream`.
 Repartition has two modes:
 
 * ``shuffle=False`` - performs the minimal data movement needed to equalize block sizes
 * ``shuffle=True`` - performs a full distributed shuffle
 
-.. image:: images/dataset-shuffle.svg
+.. image:: images/datastream-shuffle.svg
    :align: center
 
 ..
   https://docs.google.com/drawings/d/132jhE3KXZsf29ho1yUdPrCHB9uheHBWHJhDQMXqIVPA/edit
 
-Datasets can shuffle hundreds of terabytes of data. For an in-depth guide on shuffle performance, read :ref:`Performance Tips and Tuning <shuffle_performance_tips>`.
+Datastream can shuffle multi-terabyte datasets, leveraging the Ray object store for disk spilling. For an in-depth guide on shuffle performance, read :ref:`Performance Tips and Tuning <shuffle_performance_tips>`.
```

Comment:
why `datasets` here is better than `data` before?

Reply:
Lower case 'datasets' is fine to use; this was just an incidental fix to improve the wording here.